trackintel/geogr/distances.py

```diff
@@ -170,3 +169,96 @@ def meters_to_decimal_degrees(meters, latitude):
         An approximation of a distance (given in meters) in degrees.
     """
     return meters / (111.32 * 1000.0 * cos(latitude * (pi / 180.0)))
+
+
+def check_wgs_for_distance_calculation(crs):
+    """
+    This functions evaluates if the crs is wgs 84 and warns if no crs is defined.
+
+    Parameters
+    ----------
+    crs : 'pyproj.crs.crs.CRS'
+        Can be the result of `gdf.crs`
+
+    Returns
+    -------
+    is_wgs : bool
+        True if wgs84 is assumed
+
+    Notes
+    ______
+    We do not check for planar crs as this is already done when geopandas.length is called.
+
+    Examples
+    ---------
+    >>> from trackintel.geogr.distances import check_wgs_for_distance_calculation
+    >>> check_wgs_for_distance_calculation(triplegs.crs)
+    """
+
+    is_wgs = False
+
+    if crs == 4326:
+        is_wgs = True
+
+    elif crs is None:
+        is_wgs = True
+        warnings.warn('Your data is not projected. WGS84 is assumed and for length calculation the haversine '
+                      'distance is used')
+    return is_wgs
+
+
+def calc_haversine_length_of_linestrings(gdf):
```

Comment:
We have `calculate_distance_matrix` at the top, should we either use `calculate` or `calc` in both places?
Also, since linestring requirement is specified both in docstring and the assert, can be simplify this function to `calc_haversine_length`?

Reply:
In theory I would agree but in this case I would say that this function is really only useful for wgs84/haversine as you can use shapely/geopandas build-ins `linestring.length` or `gdf.length`.
So I would go with your first option.